src/main/java/io/netty/incubator/codec/quic/SegmentedDatagramPacketAllocator.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2021 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.incubator.codec.quic;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.channel.socket.DatagramPacket;
+
+import java.net.InetSocketAddress;
+
+/**
+ * Used to allocate datagram packets that use UDP_SEGMENT (GSO).
+ */
+public interface SegmentedDatagramPacketAllocator {
```

Comment:
Any reason these interfaces are not in Netty itself?

Reply:
Because usually the user hands the `DatagramPacket` to use so its easy for people to do so. While here we need to allocate internally. That said we could think about moving these at some point if it will be useful. I just would like to not do this for now

Reply:
Ok, but that means the io_uring module will depend on the QUIC module in order to make it support GSO?

Reply:
no... the user will just basically write their own implementation if they want to use it...

Reply:
same for epoll

Reply:
Okay 👍